app/controllers/email_controller.rb

```diff
@@ -24,6 +24,12 @@ def unsubscribe
 
         watching = TopicUser.notification_levels[:watching]
 
+        @unsubscribed_from_all =
+          !@user.user_option.mailing_list_mode &&
+          !@user.user_option.email_digests &&
+          @user.user_option.email_level == UserOption.email_level_types[:never] &&
+          @user.user_option.email_messages_level == UserOption.email_level_types[:never]
```

Comment:
I would extract this method into the `UserOption` model to keep the controller lean.